data/adapters/file_fileset_adapter.py

```diff
@@ -387,21 +464,92 @@ def query_fileset_files_props_igvf(self, accession):
 
+        sample_term_ids = [sample_term_id.replace(
+            ':', '_') for sample_term_id in sample_term_ids]
+        unloaded_donors, unloaded_sample_terms = self.check_hyperedges(
+            donor_ids, sample_term_ids)
+
         props = {
             '_key': accession,
             'file_set_id': fileset_accession,
             'lab': lab,
-            'preferred_assay_titles': self.none_if_empty(preferred_assay_titles),
-            'assay_term_ids': self.none_if_empty(assay_term_ids),
-            'prediction': prediction,
-            'prediction_method': prediction_method,
+            'preferred_assay_titles': preferred_assay_titles,
+            'assay_term_ids': assay_term_ids,
+            'method': method,
+            'class': class_type,
             'software': self.none_if_empty(software),
-            'samples': self.none_if_empty(sample_term_ids),
+            'samples': [f'ontology_terms/{sample_term_id}' for sample_term_id in sample_term_ids] if sample_term_ids else None,
             'sample_ids': self.none_if_empty(sample_ids),
             'simple_sample_summaries': self.none_if_empty(simple_sample_summaries),
-            'donor_ids': self.none_if_empty(donor_ids),
-            'treatments_term_ids': sorted(list(treatment_ids)) if treatment_ids else None,
+            'donors': [f'donors/{donor_id}' for donor_id in donor_ids] if donor_ids else None,
+            'treatments_term_ids': self.none_if_empty(treatment_ids),
             'publication': publication_id,
             'source': 'IGVF'
         }
-        return props
+        if replace:
+            return props, donor_ids, sample_term_ids
+        else:
+            return props, unloaded_donors, unloaded_sample_terms
+
+    def get_donor_props(self, donors, portal_url, source):
+        for donor in donors:
+            donor_url = f'{portal_url}{donor}/@@embedded?format=json'
+            donor_object = requests.get(donor_url).json()
+
+            accession = donor_object['accession']
+            sex = donor_object.get('sex', None)
+            age = donor_object.get('age', None)
+            age_units = donor_object.get('age_units', None)
+
+            if source == 'IGVF':
+                phenotypic_features = donor_object.get(
+                    'phenotypic_features', [])
+                phenotypic_feature_ids = self.none_if_empty([
+                    f"ontology_terms/{pf['feature']['term_id'].replace(':', '_')}"
+                    for pf in phenotypic_features
+                ])
+                phenotypic_feature_names = self.none_if_empty([
+                    pf['feature']['term_name']
+                    for pf in phenotypic_features
+                ])
+                ethnicities = self.none_if_empty(
+                    donor_object.get('ethnicities', []))
+
+            elif source == 'ENCODE':
+                health_status = donor_object.get('health_status')
+                phenotypic_feature_names = [
```

Comment:
have cases with health_status = unknown (e.g. ENCDO116AAA), should probably convert to None.
Also I feel it's not so useful to store ENOCDE phenotypes without an ontology term (e.g. ENCDO000ACV -> endometrial adenocarcinoma), might want to do some premapping for those cases.

Reply:
mapping will be a bit of work, coz the health_status on ENCODE is free text and is not very curated.